roofit/roofitcore/src/TestStatistics/RooSubsidiaryL.cxx

```diff
@@ -0,0 +1,68 @@
+// Author: Patrick Bos, Netherlands eScience Center / NIKHEF 2021
+
+/*****************************************************************************
+ * RooFit
+ * Authors:                                                                  *
+ *   WV, Wouter Verkerke, UC Santa Barbara, verkerke@slac.stanford.edu       *
+ *   DK, David Kirkby,    UC Irvine,         dkirkby@uci.edu                 *
+ *                                                                           *
+ * Copyright (c) 2000-2021, Regents of the University of California          *
+ *                          and Stanford University. All rights reserved.    *
+ *                                                                           *
+ * Redistribution and use in source and binary forms,                        *
+ * with or without modification, are permitted according to the terms        *
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)             *
+ *****************************************************************************/
+
+#include <TestStatistics/RooSubsidiaryL.h>
+#include <TestStatistics/kahan_sum.h>
+#include <RooAbsPdf.h> // for dynamic cast
+#include <RooErrorHandler.h>
+
+namespace RooFit {
+namespace TestStatistics {
+
+RooSubsidiaryL::RooSubsidiaryL(const std::string& parent_pdf_name, const RooArgSet &pdfs, const RooArgSet &parameter_set)
+   : RooAbsL(nullptr, nullptr, 0, 0, RooAbsL::Extended::No), parent_pdf_name_(parent_pdf_name)
+{
+   std::unique_ptr<TIterator> inputIter{pdfs.createIterator()};
+   RooAbsArg *comp;
+   while ((comp = (RooAbsArg *)inputIter->Next())) {
```

Comment:
`unique_ptr<TIterator>` goes in the right direction, but range-based for is gold.